<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OOP_Assignment11Class</class>
 <widget class="QMainWindow" name="OOP_Assignment11Class">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1162</width>
    <height>524</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OOP_Assignment11</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>1149</width>
      <height>442</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QListWidget" name="turretsList">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QFormLayout" name="formLayout">
             <property name="horizontalSpacing">
              <number>10</number>
             </property>
             <property name="verticalSpacing">
              <number>20</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Location</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="locationLine">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Size</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="sizeLine">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_3">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Aura Level</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="auraLine">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_4">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Separate Parts</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLineEdit" name="partsLine">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_5">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Vision</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QLineEdit" name="visionLine">
               <property name="font">
                <font>
                 <family>Arial</family>
                 <pointsize>12</pointsize>
                </font>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="addTurretButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="deleteTurretButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="updateTurretButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Update</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="saveTurretButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="nextTurretButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Next</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="allCriteriaButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Filter</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="mainPathLabel">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Main List Path</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="mainListPath">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="changeMainPathButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Change Path</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="mainPathLabel_2">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>MyList Path</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="myListPath">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="changeMyListPathButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Change Path</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="openButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Open MyList</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="modeChangeBox">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="currentText">
              <string>Select a mode</string>
             </property>
             <property name="currentIndex">
              <number>0</number>
             </property>
             <item>
              <property name="text">
               <string>Select a mode</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Mode A</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Mode B</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="myListWindowButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Open MyList Window</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QPushButton" name="undoButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Undo</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="redoButton">
             <property name="font">
              <font>
               <family>Arial</family>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Redo</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QListWidget" name="allCriteria">
       <property name="font">
        <font>
         <family>Arial</family>
         <pointsize>12</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1162</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="OOP_Assignment11.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>addTurretButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>addTurret()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>119</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteTurretButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>removeTurret()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>178</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>updateTurretButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>updateTurret()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>241</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>241</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveTurretButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>saveTurret()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextTurretButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>nextTurret()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>479</y>
    </hint>
    <hint type="destinationlabel">
     <x>453</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>changeMyListPathButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>changeMyListPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>891</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>749</x>
     <y>269</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>changeMainPathButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>changeMainPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>891</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>749</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modeChangeBox</sender>
   <signal>activated(QString)</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>changeMode(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>699</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>567</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>678</x>
     <y>422</y>
    </hint>
    <hint type="destinationlabel">
     <x>582</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>876</x>
     <y>422</y>
    </hint>
    <hint type="destinationlabel">
     <x>749</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>myListWindowButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>openWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>766</x>
     <y>265</y>
    </hint>
    <hint type="destinationlabel">
     <x>670</x>
     <y>312</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>allCriteriaButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>filterCriteria()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>445</x>
     <y>463</y>
    </hint>
    <hint type="destinationlabel">
     <x>490</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openButton</sender>
   <signal>clicked()</signal>
   <receiver>OOP_Assignment11Class</receiver>
   <slot>openMyList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>532</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addTurret()</slot>
  <slot>removeTurret()</slot>
  <slot>updateTurret()</slot>
  <slot>nextTurret()</slot>
  <slot>changeMainPath()</slot>
  <slot>saveTurret()</slot>
  <slot>changeMyListPath()</slot>
  <slot>changeMode(QString)</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>openWindow()</slot>
  <slot>filterCriteria()</slot>
  <slot>openMyList()</slot>
 </slots>
</ui>
